cli/src/main/kotlin/commands/ReporterCommand.kt

```diff
@@ -147,7 +148,9 @@ class ReporterCommand : CliktCommand(
         reports.forEach { (reporter, file) ->
             @Suppress("TooGenericExceptionCaught")
             try {
-                reporter.generateReport(file.outputStream(), input)
+                val outputStream = ByteArrayOutputStream()
+                reporter.generateReport(outputStream, input)
+                outputStream.writeTo(file.outputStream())
```

Comment:
Doesn't this contradict the idea of passing an output stream to the reporter? Because IIRC there were two reasons to do this, one is that the reporter does not have to care about files, and the other was that the report data can be written to the disk immediately and does not have to stay in memory for large reports like e.g. the notice by package report for large projects.
An alternative would be to delete the file in the catch block below, maybe only if it is empty because the partially written content might be useful for debugging.
Btw, thanks for looking at this, it is annoying me for a long time already.

Reply:
I like the idea of deleting the (empty) file in the `catch` block. But I'm uncertain about how to handle non-empty files in the `catch` block: On the one hand they might be useful for debugging like you say, on the other hand they might be mistakenly regarded as complete (e.g. if it happens so that a `NOTICE` file was truncated exactly after a license).

Reply:
I went for deleting the file only if it is empty.

Reply:
As an idea, we could only keep the file if `--debug` is enabled?

Reply:
Hmm, so far `--debug` is only documented to influence the log level. The only place where we already are inconsistent in this regard is `ScanCode.kt` where we pass debug options to ScanCode if `--debug` is given. But I'd rather remove that inconsistency than introducing another one; or we should document `--debug` differently. What do you prefer?

Reply:
Let's keep it as is.